script/echoclient.py

```diff
@@ -0,0 +1,38 @@
+import socket
+import sys
+
+ser_port = 1234
+cli_port = 4321
+
+def client(ip):
+    server_address = (ip, ser_port)
+    sock = socket.socket(
+        socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_TCP
```

Comment:
IPPROTO_TCP is explicitly specified here. I'd recommend either also explicitly specifying it in the server, or you can remove it and make it implicit here.